datadog.go

```diff
@@ -118,45 +102,50 @@ func (e *Exporter) Export(ctx context.Context, cs export.CheckpointSet) error {
 				if err != nil {
 					return fmt.Errorf("error getting MinMaxSumCount value for %s: %w", name, err)
 				}
-				e.client.Gauge(rec.name, metricValue(r.Descriptor().NumberKind(), val), tags, rate)
+				err = e.client.Gauge(rec.name, metricValue(r.Descriptor().NumberKind(), val), tags, rate)
+				if err != nil {
+					return err
+				}
 			}
 		case aggregator.Sum:
 			val, err := agg.Sum()
 			if err != nil {
 				return fmt.Errorf("error getting Sum value for %s: %w", name, err)
 			}
-			e.client.Count(name, val.AsInt64(), tags, rate)
+			err = e.client.Count(name, val.AsInt64(), tags, rate)
+			if err != nil {
+				return err
+			}
 		case aggregator.LastValue:
 			val, _, err := agg.LastValue()
 			if err != nil {
 				return fmt.Errorf("error getting LastValue for %s: %w", name, err)
 			}
-			e.client.Gauge(name, metricValue(r.Descriptor().NumberKind(), val), tags, rate)
+			err = e.client.Gauge(name, metricValue(r.Descriptor().NumberKind(), val), tags, rate)
+			if err != nil {
+				return err
+			}
 		}
 		return nil
 	})
 }
 
+func extractTags(r export.Record) []string {
+	itr := r.Labels().Iter()
+	tags := make([]string, itr.Len())
+	for itr.Next() {
+		i, label := itr.IndexedLabel()
+		tags[i] = string(label.Key) + ":" + label.Value.Emit()
+	}
+	return tags
+}
+
 // Close cloess the underlying datadog client which flushes
 // any pending buffers
 func (e *Exporter) Close() error {
 	return e.client.Close()
 }
 
-// sanitizeMetricName formats the custom namespace and view name to
-// Datadog's metric naming convention
-func (e *Exporter) sanitizeMetricName(namespace, name string) string {
```

Comment:
This was copied from the official OpenCensus exporter, are we sure it's okay to remove? For reference: https://github.com/DataDog/opencensus-go-exporter-datadog/blob/master/datadog.go#L119

Reply:
I don’t trust that code, no offence to open census. Periods should not be stripped for example. I just needed something quick, lean and as little garbage collection overhead as possible.

Reply:
Sounds good to me :)

Reply:
FWIW, it's not OpenCensus, it's the DataDog team that wrote that code (since it's the exporter for OpenCensus)